### PR TITLE
field import: preserve generic info

### PR DESCRIPTION
**Pull request by cal101**

### CODE REVIEW

File: src/test/java/org/walkmod/javalang/compiler/SymbolVisitorAdapterTest.java [hunk 1]
```diff
@@ -2162,4 +2162,49 @@ public void testErrorWithStaticAndNotStaticImports() throws Exception{
...
+			+ " public class Column<T> {\n"
+			+ "  public static final Column<Long> Maxtime = new Column<Long>();\n"
+			+ " }\n";
+
+	@Test
+	public void testTypeOfGenericConstant() throws Exception {
+		CompilationUnit cu = run(genericColumnAccessor);
+
+		Assert.assertTrue(true);
+		//SymbolVisitorAdapterAssertions.assertTypeOfGenericConstant(cu);
```
Comment (rpau) [inline src/test/java/org/walkmod/javalang/compiler/SymbolVisitorAdapterTest.java]: Why are these lines of code commented? Could you add the appropriate assertions in each test case?

**Review by rpau — CHANGES_REQUESTED** [review]
Thanks for the contribution! However, some assertions on the tests make describe better which parts needs to be properly evaluated.

File: src/test/java/org/walkmod/javalang/compiler/SymbolVisitorAdapterTest.java (hunk 1 above)
Comment (cal101) [reply src/test/java/org/walkmod/javalang/compiler/SymbolVisitorAdapterTest.java]: These are the fluent assertions i wrote about via email. Useful but work in progress. Details in email.

Comment (rpau) [reply src/test/java/org/walkmod/javalang/compiler/SymbolVisitorAdapterTest.java]: 👍

Comment (rpau) [reply src/test/java/org/walkmod/javalang/compiler/SymbolVisitorAdapterTest.java]: @cal101 can uncomment the code now?

**Comment by cal101** [comment]
OK, will take a look at the weekend.

**Comment by cal101** [comment]
Found some rest of energy. Should be done.

**Comment by cal101** [comment]
Fixed merge conflict. Should be ready for merge now.

File: src/main/java/org/walkmod/javalang/compiler/actions/LoadStaticImportsAction.java [hunk 2]
```diff
@@ -157,20 +158,19 @@ public void doPush(Symbol<?> symbol, SymbolTable table) throws Exception {
...
-                                st.setParameterizedTypes(null);
-                                Symbol<?> s = new Symbol(field.getName(), st, n, ReferenceType.VARIABLE, true, null);
-                                table.pushSymbol(s);
-                            }
-                        }
-                    }
-                }
-            }
-        }
+							boolean isVisible = Modifier.isPublic(modifiers)
```
Comment (rpau) [inline src/main/java/org/walkmod/javalang/compiler/actions/LoadStaticImportsAction.java]: This code follows a different format than you suggested to me. Could you reformat the code?

**Comment by cal101** [comment]
Sorry. Will take a look at all pull requests when I am back.

**Comment by cal101** [comment]
I see that formatting of the master is not clean, too but I also see that a reformatting changes things I intended to be left alone. 
The reformatting reduces readability of some tests.
I will take a look at the formatter config when I am back.
I am sorry for that.

**Comment by cal101** [comment]
reformatted and rebased on master.
But please consider accepting https://github.com/rpau/javalang-compiler/pull/66 and applying the formatter.

**Comment by cal101** [comment]
Rebased.

**Comment by cal101** [comment]
Rebased again.

**Comment by cal101** [comment]
Please tell if you have open issues. I consider it ready.

**Review by rpau — APPROVED** [review]

**Comment by rpau** [comment]
Thanks @cal101

**Pull request merged**